### PR TITLE
Migrate GTItembars to use forge's ItemDecorations rather than a mixin

### PR DESCRIPTION
**Pull request by Thepigcat76**
## What
This pr changes GT(Item/Tool)Bars to use the IItemDecorator interface for rendering and registers it in the ClientProxy. This removes the GuiGraphics mixin and makes the forge devs happy :)

## Outcome
GT(Item/Tool)Bar rendering is exactly the same as before but now uses ItemDecorators

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/client/ClientProxy.java [hunk 1]
```diff
@@ -56,6 +59,15 @@ public static void onRegisterEntityRenderers(EntityRenderersEvent.RegisterRender
         event.registerBlockEntityRenderer(GTBlockEntities.GT_SIGN.get(), SignRenderer::new);
         event.registerBlockEntityRenderer(GTBlockEntities.GT_HANGING_SIGN.get(), HangingSignRenderer::new);
     }
+    @SubscribeEvent
+    public static void onRegisterItemDecorations(RegisterItemDecorationsEvent event) {
+        for (Item item : BuiltInRegistries.ITEM) {
+            if (item instanceof IGTTool || item instanceof IComponentItem) {
+                event.register(item, new GTItemBarRenderer());
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/client/ClientProxy.java]: IMO this should use a static instance of GTItemBarRenderer, or is that not possible?

**Review by screret — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregtechceu/gtceu/client/ClientProxy.java (hunk 1 above)
Comment (Thepigcat76) [reply src/main/java/com/gregtechceu/gtceu/client/ClientProxy.java]: Changed it to use one :+1:

**Review by screret — APPROVED** [review]

**Comment by screret** [comment]
Actually you need to apply spotless first

**Comment by Thepigcat76** [comment]
you mean the formatting? Oki

**Comment by screret** [comment]
Yuh

**Pull request merged**